src/System.Reflection.Metadata.ILDasm/src/Decoder/ILDecoder.cs

```diff
@@ -0,0 +1,582 @@
+        /// <returns>true if is a user string false if not</returns>
+        public static bool IsUserString(int token)
+        {
+            return (token >> 24) == 0x70;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a member reference token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a member reference false if not</returns>
+        public static bool IsMemberReference(int token)
+        {
+            return (token >> 24) == 0x0a;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a method specification token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a method specification false if not</returns>
+        public static bool IsMethodSpecification(int token)
+        {
+            return (token >> 24) == 0x2b;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a method definition token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a method definition false if not</returns>
+        public static bool IsMethodDefinition(int token)
+        {
+            return (token >> 24) == 0x06;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a field definition token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a field definition false if not</returns>
+        public static bool IsFieldDefinition(int token)
+        {
+            return (token >> 24) == 0x04;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a type specification token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a type specification false if not</returns>
+        public static bool IsTypeSpecification(int token)
+        {
+            return (token >> 24) == 0x1b;
+        }
+
+        /// <summary>
+        /// Method that given a token defines if it is a standalone signature token.
+        /// </summary>
+        /// <param name="token">token to solve</param>
+        /// <returns>true if is a standalone signature false if not</returns>
+        public static bool IsStandaloneSignature(int token)
+        {
+            return (token >> 24) == 0x11;
+        }
+
+        public static string GetByteArrayString(byte[] array)
+        {
+            StringBuilder sb = new StringBuilder();
+            for (int i = 0; i < array.Length; i++)
+            {
+                if (i == 0)
+                {
+                    sb.Append("( ");
+                }
+                sb.Append(array[i].ToString("X2"));
+                sb.Append(" ");
+            }
+            sb.Append(")");
```

Comment:
Assuming you do need to keep this method, there are additional optimizations that can be applied.  For example, you know there won't be more than 2 + 3*array.Length characters in the resulting string, so you could initialize the StringBuilder with that size.  You could also just use a char[] created with the right size and fill in each character, rather than needing a StringBuilder.  Plus, that'll help avoid all of the intermediate garbage strings getting generated for each ToString call.  There's a very similar example of that here: https://github.com/dotnet/corefx/blob/6504501399c6f9ea4d48296058e7e9a57e41ca21/src/System.Security.Cryptography.X509Certificates/src/Internal/Cryptography/Helpers.cs#L18-L28